tests/validate_license_expr.rs

```diff
@@ -0,0 +1,72 @@
+extern crate license_exprs;
+
+use license_exprs::validate_license_expr;
+
+macro_rules! assert_licenses_ok {
+    ($($license:expr),+ $(,)+) => {{
+        $(assert!(validate_license_expr($license).is_ok());)+
+    }}
+}
+
+macro_rules! assert_licenses_err {
+    ($($license:expr),+ $(,)+) => {{
+        $(assert!(validate_license_expr($license).is_err());)+
+    }}
+}
+
+#[test]
+fn single_license() {
+    assert_licenses_ok! {
+        "MIT",
+        "MIT ",
+        " MIT",
+        " MIT ",
+        "    MIT    ",
+        "MIT+",
```

Comment:
While this is technically legal with the SPDX license expression syntax, I'd like to have future SPDX versions be able to error when `+` is applied to a license for which it makes no sense.  I [don't think the SPDX License List has any non-deprecated identifiers where `+` makes sense][1], and [`GPL-3.0+`][2] and similar are deprecated short identifiers even without a `+` operator.  So for testing `+`, I'd prefer using `AGPL-1.0+` (and `AGPL-1.0 +` where you have `MIT +` below).  The `AGPL-1.0` identifier is [deprecated in 3.1][3], but it exercises the `+` operator in a way that makes legal sense.

[1]: https://lists.spdx.org/pipermail/spdx-tech/2018-April/003574.html
[2]: https://github.com/rust-lang-nursery/license-exprs/blob/68e383192fc7781aad09b22e6612bed0e1c2ad0d/src/spdx.rs#L173
[3]: https://github.com/spdx/license-list-XML/blob/v3.1/src/AGPL-1.0.xml#L5

Reply:
I'm going to have to read a bit more around this, and see what other libraries are doing. I agree that `MIT+` doesn't make sense, but as you say it's technically legal.

I'm thinking about how to add warnings to the API as well, for things like deprecated identifiers, so adding a `+` onto a license where it doesn't make sense could be a warning until the spec explicitly denies it.

Reply:
> I'm thinking about how to add warnings to the API as well, for things like deprecated identifiers, so adding a + onto a license where it doesn't make sense could be a warning until the spec explicitly denies it.

That would be great (links to similar comments from [here][1]).

On the `MIT+` vs. `AGPL-1.0+` front, have you settled on a preference?  I prefer `AGPL-1.0+` because I weight "makes legal sense" more than "based on a non-deprecated identifier".  But I don't feel stronlgy enough about it to hold up this PR, which looks good to me otherwise.

[1]: https://github.com/rust-lang-nursery/license-exprs/pull/5#discussion_r159742256